helix-rest/src/main/java/org/apache/helix/rest/metadatastore/ZkMetadataStoreDirectory.java

```diff
@@ -339,4 +348,22 @@ public synchronized void close() {
     _routingDataMap.clear();
     _zkMetadataStoreDirectoryInstance = null;
   }
+
+  /**
+   * Make sure the root routing data path exists. Also, register the routing ZK address.
+   * @param zkClient
+   */
+  private void createRoutingDataPath(HelixZkClient zkClient, String zkAddress) {
+    try {
+      zkClient.createPersistent(MetadataStoreRoutingConstants.ROUTING_DATA_PATH, true);
+    } catch (ZkNodeExistsException e) {
+      // The node already exists and it's okay
+    }
+    // Make sure ROUTING_DATA_PATH is mapped to the routing ZK so that FederatedZkClient used
+    // in Helix REST can subscribe to the routing data path
+    ZNRecord znRecord = new ZNRecord(MetadataStoreRoutingConstants.ROUTING_DATA_PATH);
```

Comment:
The path contains"/" symbol. Would that be OK?

Reply:
That's okay. This is just an ID - it's for display purposes only.

Reply:
That was confusing user. In our convention, the path to a certain will be "/ID of ZNode1/ID of ZNode2".  Usually we dont have folder name is one thing and id of ZNRecord is another thing. Suggest to make it consisnt.